README.md

```diff
@@ -10,268 +10,475 @@ hard-to-find defects early in the application development lifecycle.
 
-```
-cd /path/to/dir && codesonar analyze ${PROJNAME} -foreground ${HUB} sh -xe mybuildscript.sh
+```bash
+cd /myfiles/src/projX && codesonar analyze $CSONAR_CMD_ARGS make normal
 ```
 
-2. Collect the build step text into a single batch file path\to\dir\mybuildbat.bat:
+### Example 2: C/C++ project; Pipeline build steps include multiple commands that involve C/C++ compilation.
+
+Suppose that the Jenkins build step text is:
 
 ```bash
-cd \myfiles\src\projX
+cd /myfiles/src/projX
 rm -f *.o
 gcc -c A.c
 gcc -c B.c
 gcc -c C.c
 ```
-then replace the build step text with:
 
-```bash
-codesonar analyze ${PROJNAME} -foreground ${HUB} path\to\dir\mybuildbat.bat
-```
-
-#### Example 3: Java project
+There are several possible approaches.
 
-Suppose that the Jenkins job writes Java build output to /myfiles/buildoutput/classes.
+* **Option 1.** Replace the build step text with:
+   ```bash
+   cd /myfiles/src/projX
+   rm -f *.o
+   codesonar build $CSONAR_CMD_ARGS gcc -c A.c
+   codesonar build $CSONAR_CMD_ARGS gcc -c B.c
+   codesonar build $CSONAR_CMD_ARGS gcc -c C.c
+   codesonar analyze $CSONAR_CMD_ARGS
+   ```
+* **Option 2.** Collect the build step text into a single shell script `/path/to/dir/mybuildscript.sh`:
+   ```bash
+   cd /myfiles/src/projX
+   rm -f *.o
+   gcc -c A.c
+   gcc -c B.c
+   gcc -c C.c
+   ```
+
+   then replace the build step text with:
+   ```bash
+   cd /path/to/dir && codesonar analyze $CSONAR_CMD_ARGS sh -xe mybuildscript.sh
+   ```
+* **Option 3.** Collect the build step text into a single batch file `path\to\dir\mybuildbat.bat`:
+   ```bash
+   cd \myfiles\src\projX
+   rm -f *.o
+   gcc -c A.c
+   gcc -c B.c
+   gcc -c C.c
+   ```
+
+   then replace the build step text with:
+
+   ```bash
+   codesonar analyze $CSONAR_CMD_ARGS path\to\dir\mybuildbat.bat
+   ```
+
+### Example 3: Java project
+
+Suppose that the Jenkins build writes Java build output to `/myfiles/buildoutput/classes`.
 
 Then add a new "Execute shell" build step with the following contents.
 
 ```bash
-codesonar analyze /myfiles/csonar_projects/projX/${PROJNAME} -foreground ${HUB} cs-java-scan /myfiles/buildoutput/classes
+codesonar analyze $CSONAR_CMD_ARGS cs-java-scan /myfiles/buildoutput/classes
 ```
 
-#### Example 4: Mixed C/C++ and Java project; single build command
+### Example 4: Mixed C/C++ and Java project; single build command
 
-Suppose that the Jenkins job build step text is:
+Suppose that the Jenkins build step text is:
 
 ```bash
 cd /myfiles/src/projX
 make all
 ```
 
-and that the Jenkins job writes Java build output to `/myfiles/buildoutput/classes`.
+and that it writes Java build output to `/myfiles/buildoutput/classes`.
 
-#### Example 5: Mixed C/C++ and Java project; multiple build commands
+Then replace the build step text with:
 
-Suppose the Jenkins job build step text is:
+```bash
+cd /myfiles/src/projX
+codesonar build $CSONAR_CMD_ARGS make all
+codesonar build $CSONAR_CMD_ARGS cs-java-scan /myfiles/buildoutput/classes
```

Comment:
I think you must use at least the `-include-artifacts` flag with `cs-java-scan`, unless that has been changed in a recent release?